libraries/ui-library/src/components/six-header/six-header.tsx

```diff
@@ -90,141 +87,159 @@ export class SixHeader {
+  sixAppSwitcherSelect!: EventEmitter<SixHeaderAppSwitcherSelectPayload>;
 
   /** Emitted when a menu item is in the profile menu is selected. */
-  @Event({ eventName: 'six-header-profile-select' })
-  sixProfileSelect: EventEmitter<SixHeaderProfileSelectPayload>;
+  @Event({ eventName: 'six-header-profile-select' }) sixProfileSelect!: EventEmitter<SixHeaderProfileSelectPayload>;
 
   /** Emitted when the hamburger menu is clicked. */
-  @Event({ eventName: 'six-header-hamburger-menu-clicked' })
-  sixHamburgerClick: EventEmitter<EmptyPayload>;
+  @Event({ eventName: 'six-header-hamburger-menu-clicked' }) sixHamburgerClick!: EventEmitter<EmptyPayload>;
 
   /** Emitted when the header logo is clicked. */
-  @Event({ eventName: 'six-header-logo-clicked' })
-  sixLogoClick: EventEmitter<EmptyPayload>;
+  @Event({ eventName: 'six-header-logo-clicked' }) sixLogoClick!: EventEmitter<EmptyPayload>;
 
   /** Emitted when search field is toggled. */
   @Event({ eventName: 'six-header-search-field-toggle' })
-  sixSearchFieldToggle: EventEmitter<SixHeaderSearchFieldToggle>;
+  sixSearchFieldToggle!: EventEmitter<SixHeaderSearchFieldToggle>;
 
-  @Element() host: HTMLSixHeaderElement;
+  @Element() host!: HTMLSixHeaderElement;
 
-  readonly eventListeners = new EventListeners();
+  private eventListeners = new EventListeners();
 
-  slots: { readonly [K: string]: boolean };
+  private slots?: Record<Slot, boolean>;
 
-  @State() selectedApp: string;
-  @State() selectedSection: Section;
+  @State() selectedApp?: string;
+  @State() selectedSection?: Section;
 
-  has = (slot: Slot) => this.slots[slot];
+  private hasSlot(slot: Slot): boolean {
+    if (this.slots == null) {
+      return false;
+    }
+    return this.slots[slot];
+  }
 
-  selected = (value: Section) => value === this.selectedSection;
+  private isSectionSelected(value: Section) {
+    return value === this.selectedSection;
+  }
 
-  select = (value: Section) => () => {
-    this.selectedSection = value;
-  };
+  private selectSection(section: Section) {
+    return () => (this.selectedSection = section);
+  }
 
-  toggleSearch = (value: Section) => () => {
-    if (value === Section.Search) {
-      const visible = this.selectedSection !== value;
-      if (visible) {
-        this.host.querySelector(`[slot="${Slot.Search}"]`).shadowRoot.querySelector('six-input')?.setFocus();
-      }
-      this.sixSearchFieldToggle.emit({ visible });
+  private toggleSearch() {
+    const visible = this.selectedSection !== Section.Search;
+    this.selectedSection = this.selectedSection === Section.Search ? Section.None : Section.Search;
+    this.sixSearchFieldToggle.emit({ visible });
+    if (this.selectedSection === Section.Search) {
+      // setFocus deferred due to https://github.com/ionic-team/stencil/issues/3772
+      setTimeout(async () => {
+        const slot = getSlot(this.host, Slot.Search);
+        slot?.shadowRoot?.querySelector('six-input')?.setFocus();
+      }, 50);
```

Comment:
Hm, not a fan of this arbitrary timeouts,. Value 0 would not work? With `50` I would be afraid that it might still be an issue no slower devices, no?